applications/ContactStructuralMechanicsApplication/tests/cpp_tests/conditions/test_mesh_tying_condition.cpp

```diff
@@ -0,0 +1,416 @@
+// }
+
+/**
+ * @brief Set boundary conditions for the given model part.
+ * @param rModelPart the model part for which to set the boundary conditions
+ */
+void SetBC(ModelPart& rModelPart)
+{
+    // Set initial solution
+    for (auto& r_node : rModelPart.Nodes()) {
+        if (r_node.X() < 0.01) {
+            r_node.FastGetSolutionStepValue(TEMPERATURE) = 1.0;
+            r_node.FastGetSolutionStepValue(TEMPERATURE, 1) = 1.0;
+            r_node.FastGetSolutionStepValue(TEMPERATURE, 2) = 1.0;
+        } else {
+            r_node.FastGetSolutionStepValue(TEMPERATURE) = 0.0;
+            r_node.FastGetSolutionStepValue(TEMPERATURE, 1) = 0.0;
+            r_node.FastGetSolutionStepValue(TEMPERATURE, 2) = 0.0;
+        }
+    }
+
+    // Fix dofs
+    for (auto& r_node : rModelPart.Nodes()) {
+        if (r_node.X() < 0.01 || r_node.X() > 0.99) {
+            r_node.Fix(TEMPERATURE);
+        }
+    }
+}
+
+/**
+ * @brief Check the solutions of the given ModelPart.
+ * @param rModelPart The ModelPart to check the solutions of.
+ */
+void CheckSolution(ModelPart& rModelPart)
+{
+    // Check solutions
+    const double tolerance = 1.0e-6;
+    for (auto& r_node : rModelPart.Nodes()) {
+        if (r_node.X() < 0.01) {
+            KRATOS_CHECK_NEAR(r_node.FastGetSolutionStepValue(TEMPERATURE), 1.0, tolerance);
+        } else if (r_node.X() > 0.99) {
+            KRATOS_CHECK_NEAR(r_node.FastGetSolutionStepValue(TEMPERATURE), 0.0, tolerance);
+        } else {
+            KRATOS_CHECK_NEAR(r_node.FastGetSolutionStepValue(TEMPERATURE), 0.5, tolerance);
+        }
+    }
+}
+
+// /**
+//  * @brief Generates a the simplest reference model part.
+//  * @param rModelPart the model part to generate the reference model for
+//  */
+// void GenerateReferenceSimplestModelPart(ModelPart& rModelPart)
+// {
+//     // Adding variable
+//     rModelPart.AddNodalSolutionStepVariable(TEMPERATURE);
+
+//     // Creating properties
+//     auto p_prop = rModelPart.CreateNewProperties(1, 0);
+
+//     // Set conductivity
+//     p_prop->SetValue(CONDUCTIVITY, 1.0);
+
+//     // Creating nodes 
+//     auto pnode1 = rModelPart.CreateNewNode(1, 0.0, 0.5, 0.0);
+//     auto pnode2 = rModelPart.CreateNewNode(2, 0.5, 0.0, 0.0);
+//     auto pnode3 = rModelPart.CreateNewNode(3, 0.5, 1.0, 0.0);
+//     auto pnode4 = rModelPart.CreateNewNode(4, 1.0, 0.5, 0.0);
+
+//     /// Add dof
+//     for (auto& r_node : rModelPart.Nodes()) {
+//         r_node.AddDof(TEMPERATURE, REACTION_FLUX);
+//     }
+
+//     // Creating elements
+//     GeometryType::Pointer p_elem_geom1 = Kratos::make_shared<Triangle2D3<Node>>(PointerVector<Node>{std::vector<Node::Pointer>({pnode1, pnode2, pnode3})});
+//     rModelPart.AddElement(Kratos::make_intrusive<TestLaplacianElement>( 1, p_elem_geom1, p_prop));
+//     GeometryType::Pointer p_elem_geom2 = Kratos::make_shared<Triangle2D3<Node>>(PointerVector<Node>{std::vector<Node::Pointer>({pnode2, pnode4, pnode3})});
```

Comment:
why so much commented code?

Reply:
It is the reference solution without mesh tying